src/components/NewArticleForm/NewArticleForm.jsx

```diff
@@ -0,0 +1,49 @@
+import { useNewArticleForm } from './useNewArticleForm';
+import { useNewArticleCreating } from './useNewArticleCreating';
+import { TextField, Button } from '@mui/material';
+import React from 'react';
+
+export const NewArticleForm = ({ article, isEditing, onClose }) => {
```

Comment:
This component is called `NewArticleForm`, but it also allows for editing. Please figure out a better name.